.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -338,7 +362,7 @@ kernel_matrix_testing_run_tests_x64:
 kernel_matrix_testing_run_tests_arm64:
   extends:
     - .kernel_matrix_testing_run_tests
-  needs: ["kernel_matrix_testing_setup_env", "upload_dependencies_arm64", "upload_system_probe_tests_arm64", "upload_minimized_btfs_arm64"]
+  needs: ["kernel_matrix_testing_setup_env", "upload_dependencies_arm64", "upload_system_probe_tests_x64", "upload_system_probe_tests_arm64", "upload_minimized_btfs_arm64"]
```

Comment:
❓ why does this `arm64` step would need a dependency on an `x64` job?

Reply:
Good catch, fixed. I tried to find out where that came in, but because I squashed I lost that history.

Reply:
Turns out this was required. It is because the e2e runner image is only `amd64`, so we need the `connector` program built for that architecture regardless if the metal+microvms will be `arm64`.

Reply:
Could it deserve a comment?

Reply:
I fixed it, so no comment needed. 😄